.github/workflows/gh-pages.yml

```diff
@@ -7,12 +7,26 @@ on:
   # Runs every time master branch is updated
   push:
     branches: [ master ]
+    # Only if docs-related files are touched
+    paths:
+      - components/eamxx/mkdocs.yml
+      - components/eamxx/docs/*
+      - components/eamxx/cime_config/namelist_defaults_scream.xml
   # Runs every time a PR is open against master
   pull_request:
     branches: [ master ]
+    # Only if docs-related files are touched
+    paths:
+      - components/eamxx/mkdocs.yml
+      - components/eamxx/docs/*
+      - components/eamxx/cime_config/namelist_defaults_scream.xml
 
   workflow_dispatch:
 
+concurrency:
+  # Prevent 2+ copies of this workflow from running concurrently
```

Comment:
Note: I'm not 100% sure this is needed, but I think so. In particular, I am afraid that, when pushing to gh-pages to deploy a pr preview, we would have a push reject if for some reason another PR had updated gh-pages after this PR had cloned.

There's also the option of canceling running jobs if the ref is updated, but I didn't have time to test that feature.